### PR TITLE
New ESLint plugin

### PR DESCRIPTION
**Pull request by matejchalk**
This PR adds an `@rx-angular/eslint-plugin` package with 10 custom ESLint rules (for reactive, performant and Zone-less Angular apps).

The rules in question were previously implemented by @Karnaukhov-kh in TSLint (this original implementation isn't open-source). I then reimplemented them in ESLint (since TSLint is deprecated), added unit tests, extended @Karnaukhov-kh's docs with examples and generally configured the ESLint plugin for public use (tested integration with `npm link`).

### CODE REVIEW

File: libs/eslint-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+```bash
+npm install --save-dev @rx-angular/eslint-plugin
+```
+
+To use the recommended ruleset, extend the configuration in your ESLint config (e.g. `.eslintrc.json`):
+
+```json
+{
+  "extends": ["@rx-angular/recommended"]
```
Comment (hoebbelsB) [inline libs/eslint-plugin/README.md:23]: i love that one :)

File: libs/eslint-plugin/src/lib/rules/prefer-no-lodash-clone-deep.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  name: path.parse(__filename).name,
+  meta: {
+    docs: {
+      recommended: 'error',
+      description: 'Detects all usages of cloneDeep from Lodash.',
+    },
+    type: 'problem',
+    messages: {
+      [MESSAGE_ID]:
+        'Use with caution - cloneDeep is a costly operation that may cause performance problems.',
```
Comment (hoebbelsB) [inline libs/eslint-plugin/src/lib/rules/prefer-no-lodash-clone-deep.ts:18]: we could give a hint here and point to `structuredClone` https://web.dev/structured-clone/

**Review by hoebbelsB — APPROVED** [review]
@matejchalk amazing PR, big big thx for the contribution! I hope we can soon move forward with our release-schedule!

File: libs/eslint-plugin/src/lib/rules/prefer-no-lodash-clone-deep.ts (hunk 2 above)
Comment (matejchalk) [reply libs/eslint-plugin/src/lib/rules/prefer-no-lodash-clone-deep.ts]: Good suggestion :+1: I've added it now.

**Comment by edbzn** [comment]
Hey, I love this PR! 

Here are some struggles I faced in the usage of rxa that I think we can avoid by adding proper eslint rules. 

1- The first thing I face over and over is the inconsistent mixing of reactive and imperative usage. This is a real struggle for me and I think we can prevent this by adding a rule. Here is an example, of that inconsistent mixing:

```ts
     this.state.connect(
            'something',
            this.service.something$.pipe(
                map((something) =>
                   ({ ...something, ...this.state.get('somethingElse') }) // <- please no
                ),
            ),
      );
```

The rule would be no imperative usage within `state#hold` and `state#connect` methods.

2- Another useful rule that I can see is the recommended place to put both `state#hold` and `state#connect`, I have seen some usage of these methods in another place than the constructor (eg in a method called in the template). Which IMO is bad for several reasons (not reactive).

```ts
export class MyComponent {
    doSomething() {
        this.state.connect('something', this.service.doSomething()) // <- please no
    }
}
```

3- For this one, I'm not sure if it's possible, but I often see mixed usage of state properties and component properties which leads to inconsistent component state. Eg: 

```ts
export class MyComponent {
    isLoading = false; // <- please no
    
    constructor(private state: RxState<{ loaded: boolean }>) {}
}
```

What do you think about those?

**Comment by matejchalk** [comment]
Closed this by accident :facepalm: I added 2 of @edbzn's suggested rules, you can check them out in the replacement PR #1394.

**Pull request merged**